hy/compiler.py

```diff
@@ -324,6 +324,18 @@ def is_unpack(kind, x):
             and x[0] == "unpack-" + kind)
 
 
+def make_hy_model(outer, x, rest):
+   return outer(
+      [HySymbol(a) if type(a) is str else
+              a[0] if type(a) is list else a
+          for a in x] +
+      (rest or []))
+def E(*items, **kwargs):
```

Comment:
I can understand the rationale for having these factored out and to be concise, but I'm not a fan of single-character names for global functions...

Reply:
What about `HE` and `HL`, for `HyExpression` and `HyList`?

Reply:
Ehh, I just don't want the compiler to have an overabundance of these short-named functions. I guess the primary reason this exists is just to let us write Hy to be compiled inside of the compiler, in which case IMO there are probably more elegant ways to handle it (I've been toying with some ideas that I'll probably file in a separate issue/rfc). Just...not a fan of these functions for now.

Reply:
I think only two are necessary, which doesn't seem like an overabundance to me. But to be clear, what do you need from me for this PR?

Reply:
Err true 😅 Could this change just be separate from the assert fix for now?

Reply:
But the assert fix uses the new functions. Are you asking me to rewrite it without them?

Reply:
Gah, could we just do something like `mkexpr` or similar for now, not quite as short but shorter than `HyExpression` with manually passing everything over.

Reply:
Done.